### PR TITLE
Add parameters to Set-AzPolicyAssignment, fixes

### PR DESCRIPTION
**Pull request by mentat9**
## Checklist

- [X] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [X] The title of the PR is clear and informative
- [X] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [X] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [X] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [X] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1 [hunk 1]
```diff
@@ -881,13 +908,14 @@ $policyAssignmentNotFound = 'PolicyAssignmentNotFound : '
 $policySetDefinitionNotFound = 'PolicySetDefinitionNotFound : '
 $policyDefinitionNotFound = 'PolicyDefinitionNotFound : '
 $invalidRequestContent = 'InvalidRequestContent : The request content was invalid and could not be deserialized: '
-$missingSubscription = 'MissingSubscription : The request did not have a provided subscription. All requests must have an associated subscription Id.'
+#$missingSubscription = 'MissingSubscription : The request did not have a provided subscription. All requests must have an associated subscription Id.'
```
Comment (cormacpayne) [inline src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1]: @mentat9 please delete this line if it's going to be commented-out

File: src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1 [hunk 2]
```diff
@@ -559,6 +559,18 @@ function Test-PolicyAssignmentWithParameters
     Assert-AreEqual $array[0] $expected.Properties.Parameters.listOfAllowedLocations.Value[0]
     Assert-AreEqual $array[1] $expected.Properties.Parameters.listOfAllowedLocations.Value[1]
 
+	# update parameters
+    # this is validation for https://github.com/Azure/azure-powershell/issues/6055
+    $actual = Set-AzPolicyAssignment -Name testPAWP -Scope $rg.ResourceId -PolicyParameter '{ "listOfAllowedLocations": { "value": [ "something", "something else" ] } }'
```
Comment (cormacpayne) [inline src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1]: @mentat9 it looks like new cmdlet calls were added to this test, but there's no update to the corresponding `.json` recording file -- you will need to re-record this test to ensure that the correct recordings are being used in playback

File: src/Resources/Resources/help/Set-AzPolicyAssignment.md [hunk 3]
```diff
@@ -209,6 +238,36 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -PolicyParameter
```
Comment (cormacpayne) [inline src/Resources/Resources/help/Set-AzPolicyAssignment.md]: @mentat9 you be able to add examples using the new `-PolicyParameter` and `-PolicyParameterObject` parameters?

File: src/Resources/Resources/ChangeLog.md [hunk 4]
```diff
@@ -22,6 +22,9 @@
     - Add `-SkipResourceNameParameterization` parameter to Export-AzResourceGroup
     - Add `-SkipAllParameterization` parameter to Export-AzResourceGroup
     - Add `-Resource` parameter to Export-AzResourceGroup for exported resource filtering
+    - Fix help text for Get-AzPolicyState -Top parameter
```
Comment (cormacpayne) [inline src/Resources/Resources/ChangeLog.md]: @mentat9 we just had a release, so there shouldn't be anything existing under the `## Upcoming Release` header -- I would pull the latest from the `master` branch and ensure that the only content under the `## Upcoming Release` header is the lines you are adding

File: src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1 (hunk 1 above)
Comment (mentat9) [reply src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1]: Done.

File: src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1 (hunk 2 above)
Comment (mentat9) [reply src/Resources/Resources.Test/ScenarioTests/PolicyTests.ps1]: Ah, forgot to copy those over, thanks.

File: src/Resources/Resources/help/Set-AzPolicyAssignment.md (hunk 3 above)
Comment (mentat9) [reply src/Resources/Resources/help/Set-AzPolicyAssignment.md]: Done.

File: src/Resources/Resources/ChangeLog.md (hunk 4 above)
Comment (mentat9) [reply src/Resources/Resources/ChangeLog.md]: Updated.

**Comment by mentat9** [comment]
Applied PR feedback. Also added some additional doc fixes we had pending. Please take a look and comment or approve.

**Pull request merged**